Google.Api.Generator.Tests/ProtoTests/ResourceNames/ResourceNamesClient.cs

```diff
@@ -254,6 +254,155 @@ public abstract class ResourceNamesClient
             SimpleRepeatedMethodAsync(names, gaxgrpc::CallSettings.FromCancellationToken(cancellationToken));
         // TEST_END
 
+        public Response WildcardMethod(WildcardResource request, gaxgrpc::CallSettings callSettings = null) => throw new sys::NotImplementedException();
+        public stt::Task<Response> WildcardMethodAsync(WildcardResource request, gaxgrpc::CallSettings callSettings = null) => throw new sys::NotImplementedException();
+
+        // TEST_START
+        /// <summary>
+        /// Test wilcard resource.
+        /// </summary>
+        /// <param name="name">
+        /// </param>
+        /// <param name="names">
+        /// </param>
+        /// <param name="requiredSingle">
+        /// </param>
+        /// <param name="requiredRepeated">
+        /// </param>
+        /// <param name="callSettings">If not null, applies overrides to this RPC call.</param>
+        /// <returns>The RPC response.</returns>
+        public virtual Response WildcardMethod(string name, scg::IEnumerable<string> names, string requiredSingle, scg::IEnumerable<string> requiredRepeated, gaxgrpc::CallSettings callSettings = null) =>
+            WildcardMethod(new WildcardResource
+            {
+                Name = name ?? "",
+                Names =
+                {
+                    names ?? linq::Enumerable.Empty<string>(),
+                },
+                RequiredSingle = gax::GaxPreconditions.CheckNotNullOrEmpty(requiredSingle, nameof(requiredSingle)),
+                RequiredRepeated =
```

Comment:
I'm always somewhat dubious of "required repeated" - it's not like we're checking that it's non-empty. Probably worth further consideration.

Reply:
Yes, I agree, and we may want to discuss changing how we support this.
However, this matches the existing code-gen, so is "correct".